#goal definition
char[] inputContent
---
#result definition
char[] returnMessage
---
#feedback
int32[] sequence